### DISCUSSION TITLE
Shell Script Output in Plash?

### DISCUSSION THREAD

**Discussion by fpuhan** [sindresorhus/Plash#124] [Q&A] [answered]
@sindresorhus, kindly directed me here.  Thank you!

I came across Plash as I was looking for a way to superimpose text on my Mac's desktop. The post by @sindresorhus "Show any kind of text on your desktop with the ability to drag it around" is perfect, if it could be redone to simply display the output of a sqlite3 query issued via shell script. I wish to have the file names of the images I use as wallpapers displayed as well as the images themselves I can fetch these via this command:

sqlite3 ~/Library/Application\ Support/Dock/desktoppicture.db "select * from data;" | tail -5

This database is updated every time a wallpaper image changes. I use five "spaces" on my Mac, hence the 'tail' command.

Is there a way to modify the above script to allow for using a shell script? Thanks!  I am NOT a Javascript/perl/php coder!

(P.S. I currently use the script above in an Apple Quick Action Shortcut, but that means I have to find the Services entry in an app's menu.)

![image](https://user-images.githubusercontent.com/99229315/223146860-9c7aaab7-6a1d-4bce-a1ff-2c1a26bf0748.png)

**Comment by sindresorhus** [comment, marked as answer]
Copy-paste the code from https://github.com/sindresorhus/Plash/issues/1#issuecomment-1146658131 into a `Text` action in shortcuts and then replace `Hello` with a variable for the output of your shell action. Then use the `Save File` action to write a file called `index.html` into a folder of your choosing. Then choose that folder in Plash using the "Local Website" button. After that, add the `Reload Website` actions provided by Plash.

You can use [Shortery](https://apps.apple.com/us/app/shortery/id1594183810?mt=12) to automate running the shortcut at an interval.

**Reply by fpuhan** [reply]
Thank you!  I have a preliminary working script.  The output needs formatting, since the results in HTML are simple a string of data. 

This is my Shortcut:

![image](https://user-images.githubusercontent.com/99229315/223232305-c44dfeff-1e5a-47a8-827c-6ba3bc8f5793.png)

Since it's a Shortcut and automations aren't (yet) enabled under macOS (I'm running Ventura 13.2.2 on an M1 MacBook Pro), so I'm creating a launchctl process that will run the following script on a set interval (yet to be determined):

#!/bin/zsh
#
shortcuts run "Plash"
open --background 'plash:reload'
